docs/resources/slo.md

```diff
@@ -37,7 +37,11 @@ The following arguments are supported:
 
 * `name` - (Required) The name of the SLO.
 * `description` - (Optional) A description of the SLO's intent and context.
-* `dataset` - (Required) The dataset this SLO is created in. Must be the same dataset as the SLI unless the SLI's dataset is `"__all__"`.
+* `dataset` - (Deprecated, conflicts with `datasets`) The dataset this SLO is created in. Must be the same dataset as the SLI unless the SLI's dataset is `"__all__"`.
```

Comment:
I'd rather we _not_ mark it deprecated right away: a lot of people use this resource and they'll be getting warnings until they re-create all of their SLOs

we should instead make a note that it'll be deprecated in a future release